mmv1/third_party/terraform/utils/node_config.go.erb

```diff
@@ -1063,45 +1056,72 @@ func containerNodePoolTaintSuppress(k, old, new string, d *schema.ResourceData)
 		Key, Value, Effect string
 	}
 
-	taintSet := make(map[taintType]struct{})
-
-	// Add all new taints to set.
-	for _, raw := range n.([]interface{}) {
+	// Populate a map of existing taints
+	oldTaints := make(map[taintType]struct{})
+	for _, raw := range o.([]interface{}) {
 		data := raw.(map[string]interface{})
 		taint := taintType{
 			Key:    data["key"].(string),
 			Value:  data["value"].(string),
 			Effect: data["effect"].(string),
 		}
-		taintSet[taint] = struct{}{}
+		oldTaints[taint] = struct{}{}
 	}
 
-	// Remove all current taints, skipping GKE-managed keys if not present in
-	// the new configuration.
-	for _, raw := range o.([]interface{}) {
+	// Populate the diff of new taints
+	newTaints := make(map[taintType]struct{})
+	for _, raw := range n.([]interface{}) {
 		data := raw.(map[string]interface{})
 		taint := taintType{
 			Key:    data["key"].(string),
 			Value:  data["value"].(string),
 			Effect: data["effect"].(string),
 		}
-		if _, ok := taintSet[taint]; ok {
-			delete(taintSet, taint)
-		} else if !strings.HasPrefix(taint.Key, "sandbox.gke.io/") {
-			// User-provided taint removed in new configuration.
-			return false
+		newTaints[taint] = struct{}{}
+	}
+
+	// Populate the diff of added and removed taints
```

Comment:
note to self: the "taint" here is a taintType that includes the key, value, and effect inside it, so if any of those are changed that's considered a removal + addition, not a modification.